public/js/monitor.js

```diff
@@ -95,25 +124,39 @@ async function sha1(message) {
 
 async function hashEmailAndSend(emailFormSubmitEvent) {
   emailFormSubmitEvent.preventDefault();
-  handleEvents("Scan");
   const emailForm = emailFormSubmitEvent.target;
+  if(emailForm.querySelector("input[name=email]").value === ""){
+    emailForm.querySelector("input[type=submit").disabled = true;
+    addInvalidMessage();
+    return;
+  }
+  handleEvents("Scan");
   for (const emailInput of emailForm.querySelectorAll("input[type=email]")) {
     emailForm.querySelector("input[name=emailHash]").value = await sha1(emailInput.value);
     emailInput.value = "";
   }
   emailForm.submit();
-  displayLoader();
+  addClass(".input-group-button", "loading-data");
 }
 
 if(document.querySelector(".email-scan")){
-  window.addEventListener("pageshow", removeLoader);
+  window.addEventListener("pageshow", function(){
+    removeClass(".input-group-button", "loading-data");
+  });
   document.querySelector(".email-scan").addEventListener("submit", hashEmailAndSend);
   document.querySelector(".email-to-hash").addEventListener("input", enableBtnIfEmailValid);
+  document.querySelector(".input-group-button").addEventListener("click", showMessageIfDisabled);
+  document.querySelector(".email-to-hash").addEventListener("focusout", enableBtnIfInputEmpty);
+  document.querySelector(".email-to-hash").addEventListener("focusin", function(){
```

Comment:
Not sure if you want to cache `document.querySelector(".email-to-hash")`. I think we may be calling it 3 places above ("input" handler, "focusin" handler, and "focusout" handler).

Reply:
👍